app/authentication/auth.py

```diff
@@ -47,9 +47,11 @@ def restrict_ip_sms():
     ip = ''
     if request.headers.get("X-Forwarded-For"):
         # X-Forwarded-For looks like "203.0.113.195, 70.41.3.18, 150.172.238.178"
-        ip_list = request.headers.get("X-Forwarded-For")
-        ip = ip_list.split(',')[0].strip()
-        current_app.logger.info("Inbound sms ip list {}".format(ip_list))
+        ip_route = request.headers.get("X-Forwarded-For")
+        ip_list = ip_route.split(',')
+        if len(ip_list) >= 3:
+            ip = ip_list[len(ip_list) - 3]
+        current_app.logger.info("Inbound sms ip route list {}".format(ip_route))
```

Comment:
So two things:

1. Might be worth adding a comment here to give some context as to why we're looking for the that specific IP  in the list. Although you explained it to me, just in case anyone else comes across this code they'll need some context.

2. Worth creating a test for this to check it actually works